iXBRLViewerPlugin/viewer/src/js/report.js

```diff
@@ -305,3 +308,11 @@ iXBRLReport.prototype.usesAnchoring = function() {
 iXBRLReport.prototype.hasValidationErrors = function() {
     return this.data.validation !== undefined && this.data.validation.length > 0;
 }
+
+// Get a human-readable label for the provided namespace URI, from the i18n
+// taxonomies file.
+// Returns def if none available.
+iXBRLReport.prototype.taxonomyNameForURI = function(nsURI, def) {
+    const nsURIEscaped = nsURI.replaceAll(/[.:]/g,'_');
```

Comment:
Perhaps it's worth a small comment mentioning `nsSeparator` and `keySeparator` to make it clearer why we're doing these replacements?  I'm not so familiar with i18next, so it took me a bit to dig up why they were special.